analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsShouldBeLogged.Microsoft.Extensions.Logging.cs

```diff
@@ -0,0 +1,405 @@
+        }
+    }
+
+    public void MultipleLogsInTheSameCatch()
+    {
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+            logger.LogWarning(new EventId(1), e, "Message!");
+        }
+
+        try { }
+        catch (DivideByZeroException e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+            logger.LogInformation(new EventId(1), "Message!");          // Compliant - the exception has been loged already
+        }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), e, "Message!");
+            if (true)
+            {
+                logger.LogInformation(new EventId(1), "Message!");      // Compliant - the exception has been loged already
+            }
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            logger.LogWarning(new EventId(1), "Message!"); // Noncompliant
+            logger.LogWarning(new EventId(1), "Message!"); // Noncompliant
+        }
+    }
+
+    public void UsageAsStaticMethod()
+    {
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.Log(logger, LogLevel.Critical, "Message!");    // Noncompliant
+            LoggerExtensions.LogCritical(logger, "Message!");               // Noncompliant
+            LoggerExtensions.LogDebug(logger, "Message!");                  // Noncompliant
+            LoggerExtensions.LogError(logger, "Message!");                  // Noncompliant
+            LoggerExtensions.LogInformation(logger, "Message!");            // Noncompliant
+            LoggerExtensions.LogTrace(logger, "Message!");                  // Noncompliant
+            LoggerExtensions.LogWarning(logger, "Message!");                // Noncompliant
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.Log(logger, LogLevel.Critical, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogCritical(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogDebug(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogError(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogInformation(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogTrace(logger, e, "Message!");
+        }
+
+        try { }
+        catch (Exception e)
+        {
+            LoggerExtensions.LogWarning(logger, e, "Message!");
+        }
+    }
+
+    public void NoLogsInCatch()
+    {
+        try { }
+        catch (Exception e)
+        {
+        }
+    }
+
+    public void CallingMethodToLog()
```

Comment:
Add a test case where the logging method is called inside a lambda (inside a catch block).